### PR TITLE
feat: add method to setup window reset on test hooks

### LINKED ISSUE #57: Export function to setup window mock reset in new ember-qunit tests
Thanks for maintaining this addon.

The recent version of ember-qunit [applies functions on the qunit hooks](https://github.com/emberjs/ember-qunit#setup-tests) to setup ember specific stuff:

```js
import { setupTest } from 'ember-qunit';

module('SidebarController', function(hooks) {
  setupTest(hooks);
  // ...
});
```

`ember-window-mock` could export a function like `setupWindowMockTest(hooks)` which adds calls to `reset` in the hooks beforeEach, afterEach functions.

e.g. something like:
```js
import {setupWindowMockTest} from 'ember-window-mock';

module('SidebarController', function(hooks) {
  setupWindowMockTest(hooks);
  // ...
});
```

This is mostly cosmetic but reduces the work a user has to do to use window mock in testing.
If this is an addition to window mock that you want to have, I can open a PR.

### PR DESCRIPTION
**Pull request by makepanic**
fixes #57

I'm actually not sure if the note about mocha is true.
I found that part when looking at the [mirage setup method](https://github.com/samselikoff/ember-cli-mirage/blob/master/addon-test-support/setup-mirage.js#L9).

The name and documentation isn't final. If anyone has better suggestions, please tell :smiley:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,6 +91,21 @@ import { reset } from 'ember-window-mock';
 
 This function should be called before all tests that depend on the window mock, preferably in the `beforeEach` hook. See below for some examples!
 
+### Resetting the state in with QUnit 2.0
```
Comment (simonihmig) [inline README.md]: AFAIK the new API is not really related to QUnit 2.0, rather to the RFC, aka the "new testing APIs".

File: README.md [hunk 2]
```diff
@@ -91,6 +91,21 @@ import { reset } from 'ember-window-mock';
...
+
+If your test are using the QUnit 2.0 test syntax, introduced in [RFC 0232](https://github.com/emberjs/rfcs/blob/master/text/0232-simplify-qunit-testing-api.md),
+then you can setup window mock by calling the `setupWindowMock` method:
+
+```js
+import setupWindowMock from 'ember-window-mock/setup-window-mock';
```
Comment (simonihmig) [inline README.md]: I think I would prefer a named export directly from `'ember-window-mock'` as the public export:

```
import { setupWindowMock } from 'ember-window-mock';
```

You can leave the file as it is, just reexport the function from `index.js`, similar to the existing `reset` function.

File: addon-test-support/setup-window-mock.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { reset } from 'ember-window-mock';
+
+//
+// Used to reset window-mock for a test.
+//
+// NOTE: the `hooks = self` is for mocha support
+//
+export default function setupWindowMock(hooks = self) {
+  hooks.beforeEach(reset);
+  hooks.afterEach(reset);
```
Comment (simonihmig) [inline addon-test-support/setup-window-mock.js]: Not sure, do we really need to run it before *and* after each test? Shouldn't `beforeEach` be sufficient?

File: tests/acceptance/window-test.js [hunk 4]
```diff
@@ -12,4 +14,16 @@ module('Acceptance | window', function(hooks) {
 
     assert.equal(window.location.href, 'http://www.example.com/');
   });
+
+  sinonTest('it calls reset in the hooks', function(assert) {
```
Comment (simonihmig) [inline tests/acceptance/window-test.js]: It seems this does not have to be in an acceptance test. You could move that to a dedicated unit test, like `tests/unit/setup-window-mock-test.js`.

File: README.md (hunk 2 above)
Comment (makepanic) [reply README.md]: The PR now reexports the setupWindowMock method

File: addon-test-support/setup-window-mock.js (hunk 3 above)
Comment (makepanic) [reply addon-test-support/setup-window-mock.js]: I've updated the PR to only reset in the beforeEach hook

File: tests/acceptance/window-test.js (hunk 4 above)
Comment (makepanic) [reply tests/acceptance/window-test.js]: I've moved it into a unit test file

File: README.md (hunk 1 above)
Comment (makepanic) [reply README.md]: I renamed the README headline. Not sure if it's fitting enough.

**Comment by simonihmig** [comment]
Released as 0.4.1

**Pull request merged**